pkg/jwt/basic/provider.go

```diff
@@ -0,0 +1,35 @@
+package basic
+
+import (
+	"github.com/hellofresh/janus/pkg/config"
+	"github.com/hellofresh/janus/pkg/jwt/provider"
+)
+
+func init() {
+	provider.Register("basic", Provider{})
+}
+
+// Provider abstracts the authentication for github
+type Provider struct {
+	provider.Verifier
+}
+
+// Build acts like the constructor for a provider
+func (gp Provider) Build(config config.Credentials) provider.Provider {
+	return &Provider{
+		Verifier: provider.NewVerifierBasket(
+			NewPasswordVerifier(userConfigToTeam(config.Basic.Users)),
+		),
+	}
+}
+
+func userConfigToTeam(configUser []config.BasicUsersConfig) []*User {
+	users := []*User{}
+	for _, u := range configUser {
+		users = append(users, &User{
```

Comment:
maybe it's worth to extract a constructor `NewUser(username, password)`

Reply:
It's not worth it, the user is only used by this package and only by this function. A constructor should only be used if:

- It's gonna be exposed
- It requires some contextual validation to avoid inconsistent objects

Reply:
Agreed, if it's not being consumed from outside that's right, as I saw the user struct and attributes publicly declared I thought it would be. Maybe its makes sense to declare it private then